Add config for showing username in window title

SoyChai opened the pull request:
Fixes #1918 

Solved my own issue as @rherms was working on #1951. Either solution is probably equally effective, this one just includes a config to make the behaviour optional.

![image](https://user-images.githubusercontent.com/37604308/39283294-f5ecabac-4950-11e8-8a07-f8fa366838fa.png)

rherms commented on runelite-client/src/main/java/net/runelite/client/ui/ClientUI.java:
```diff
@@ -198,6 +205,12 @@ public void onConfigChanged(ConfigChanged event)
 	}
 
+	@Subscribe
+	public void onPlayerSpawned(PlayerSpawned event)
```
Are you sure this doesn't also fire when other players "spawn"?

SoyChai replied:
I suspected the same thing, so I waited inside the GE to find out, and none of the players teleporting in were triggering this callback.

SoyChai replied:
Scratch that -- it actually *does* fire for every player nearby, I wasn't checking it properly.

rherms replied:
Sweet. I also like the config for yours. I'll close mine out. Do we know what format we want to display the username? I saw you went with "RuneLite - name" and I went with "RuneLite (name)"

SoyChai replied:
I picked a hyphen format because it works on both sides, since it looks a bit odd to have the window name start with a parenthesis. Let me take a minute to add a similar justLogged check to the one you implemented, we don't want this callback shooting off constantly for no benefit 👅

rherms commented on runelite-client/src/main/java/net/runelite/client/ui/ClientUI.java:
```diff
@@ -198,6 +210,22 @@ public void onConfigChanged(ConfigChanged event)
+	@Subscribe
+	public void onGameStateChanged(GameStateChanged event)
+	{
+		justLoggedIn = event.getGameState() == GameState.LOGGED_IN;
```
So apparently LOGGED_IN fires every time you move to a new region haha (where the map says loading and loads the new region). Not very well named at all. Pretty sure the people in the discord were saying we need to listen for certain packets that are only sent on login.

SoyChai replied:
Oof, can you remember who mentioned those packets? I'll try to prod them for more information.

Adam- replied:
To test for logged in we generally test for login screen to logged in or world hop to logged in.

rherms replied:
Oh so if last state was LOGIN_SCREEN and the new state is LOGGED_IN? I like that idea!

Adam- replied:
I would expect the username/player to be in place at the time of this event, though?

deathbeam replied:
I would just set it on CLICK_TO_PLAY screen, reset on LOGIN_SCREEN?

rherms replied:
I was just hoping we could find a way to avoid "updating" the title/user name every LOGGED_IN event. It works, but it just does more than it should and idk if there are perf implications of constantly changing the frame's title. Ideally there is an event or something that can tell us if a log in occurs, and that's the only time the event is fired.

SoyChai commented:
I thought the choice would be appreciated, because appending the username at the end will cause it to be obscured on a Windows taskbar button, especially if many windows are open.

deathbeam commented on runelite-client/src/main/java/net/runelite/client/config/RuneLiteConfig.java:
```diff
@@ -79,11 +79,22 @@ default boolean gameAlwaysOnTop()
+			description = "Show the last account you logged onto in the window title",
+			position = 5
+	)
+	default UsernameTitlePosition showUsernameInTitle()
```
Just do this like OSB does, I think they do not have toggle, so I would not add one too.

SoyChai commented:
Rebased, and rewritten. It can now recognise username changes mid-game.

I still believe it's important to provide options for this feature like I mentioned in my last comment, so that toggle is unchanged.

SoyChai commented:
Rebased on post-deob history.
